log/logger.go

```diff
@@ -42,34 +121,206 @@ func File(path string) (logger *Logger, loggerClose func(), err error) {
 		}
 	}
 
-	logger = newLoggerWithWriter(file)
+	logger = newLoggerWithWriter(file, FlagStandard)
 
 	return logger, loggerClose, nil
 }
 
 // STDOUT returns a logger that writes to STDOUT.
 func STDOUT() (logger *Logger) {
-	return newLoggerWithWriter(os.Stdout)
+	return newLoggerWithWriter(os.Stdout, FlagStandard)
 }
 
-// WithFile returns a logger that writes to a file and to the parent logger at the same time.
-func WithFile(parent *Logger, filePath string) (logger *Logger, loggerClose func(), err error) {
+// newLogWriter returns a logger that writes to a file and to the parent logger at the same time.
+func newLogWriter(parent io.Writer, filePath string) (writer io.Writer, err error) {
 	if err := os.MkdirAll(filepath.Dir(filePath), 0755); err != nil {
-		return nil, nil, pkgerrors.WithStack(err)
+		return nil, pkgerrors.WithStack(err)
 	}
 
 	file, err := os.OpenFile(filePath, os.O_APPEND|os.O_WRONLY|os.O_CREATE, 0644)
 	if err != nil {
-		return nil, nil, pkgerrors.WithStack(err)
+		return nil, pkgerrors.WithStack(err)
 	}
-	loggerClose = func() {
-		if err := file.Close(); err != nil {
-			panic(err)
+
+	writer = io.MultiWriter(parent, file)
+
+	return writer, nil
+}
+
+// spawningHandler is a structural logging handler which spawns a new log file if one of the given log file spawners triggers.
+type spawningHandler struct {
+	// writer holds the writer to write the output.
+	writer io.Writer
+
+	// attributes holds the attributes handed to the logger.
+	attributes map[AttributeKey]string
+
+	// logFileSpawners holds the spawners responsible for spawning a new log file.
+	logFileSpawners []logFileSpawner
+
+	// flags holds the flags deciding which fields are logged.
+	flags Flags
+}
+
+// newSpawningHandler returns a new spawning handler.
+func newSpawningHandler(writer io.Writer, flags Flags) *spawningHandler {
+	return &spawningHandler{
+		writer: writer,
+
+		attributes: map[AttributeKey]string{},
+
+		flags: flags,
+	}
+}
+
+// Clone returns a copy of the object.
+func (h *spawningHandler) Clone() (clone *spawningHandler) {
+	return &spawningHandler{
+		writer: h.writer,
+
+		attributes: maps.Clone(h.attributes),
+
+		logFileSpawners: slices.Clone(h.logFileSpawners),
+	}
+}
+
+var _ slog.Handler = (*spawningHandler)(nil)
+
+// Enabled reports whether the handler handles records at the given level.
+func (h *spawningHandler) Enabled(ctx context.Context, level slog.Level) bool {
+	return true
+}
+
+// Handle handles the Record.
+func (h *spawningHandler) Handle(ctx context.Context, record slog.Record) error {
+	if h.flags&FlagDate != 0 {
+		fmt.Fprint(h.writer, record.Time.Format("2006/01/02"))
+		fmt.Fprint(h.writer, " ")
+	}
+	if h.flags&FlagTime != 0 {
+		fmt.Fprint(h.writer, record.Time.Format("15:04:05"))
+		fmt.Fprint(h.writer, " ")
+	}
+
+	fmt.Fprintln(h.writer, record.Message)
+
+	return nil
+}
```

Comment:
is not handling the attributes of the record itself... right? so we always have to pass the attributes to the `handler.WithAttrs`?

would maybe use `record.NumAttrs` to check that and throw/log an error in case we get attributes here

Reply:
hmm. would not do that I think. 

if we want to log query responses we probably do something like `log.Info(response, AttributeKeyArtifact, ArtifactTypeQueryResponse)`. The attribute is here only valid for the log message, we do not need to create a separate logger for that.

Reply:
I guess calls like `log.Infof` with specific attributes are anyways not possible yet, so the attributes can never show up here.